src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -727,12 +730,75 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    bool TerminalPage::UpdatesAvailable() const
+    {
+        return _pendingUpdateVersion.size() > 0;
+    }
+
+    winrt::hstring TerminalPage::PendingUpdateVersion() const
+    {
+        return _pendingUpdateVersion;
+    }
+
+    void TerminalPage::_SetPendingUpdateVersion(const winrt::hstring& version)
+    {
+        _pendingUpdateVersion = version;
+        _PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"PendingUpdateVersion" });
+        _PropertyChangedHandlers(*this, WUX::Data::PropertyChangedEventArgs{ L"UpdatesAvailable" });
+    }
+
+    winrt::fire_and_forget TerminalPage::_QueueUpdateCheck()
+    {
+        auto strongThis = get_strong();
+        auto now{ std::chrono::system_clock::now() };
+        if (now - _lastUpdateCheck < std::chrono::days{ 1 })
+        {
+            co_return;
+        }
+        _lastUpdateCheck = now;
+
+        if (!IsPackaged())
+        {
+            co_return;
+        }
+
+        co_await wil::resume_foreground(strongThis->_tabView.Dispatcher());
+        _SetPendingUpdateVersion({});
+        CheckingForUpdates(true);
+
+        try
+        {
+#ifdef WT_BRANDING_DEV
+            co_await winrt::resume_after(std::chrono::seconds{ 3 });
+            co_await wil::resume_foreground(strongThis->_tabView.Dispatcher());
+            _SetPendingUpdateVersion(L"X.Y.Z");
+#else // release build, likely has a store context
+            auto storeContext = winrt::Windows::Services::Store::StoreContext::GetDefault();
```

Comment:
presumably
```suggestion
            if(auto storeContext = winrt::Windows::Services::Store::StoreContext::GetDefault())
            {
```

and wrap up the rest of this as a conditional